<launch>

   <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find wpr_simulation)/config/wpr1_control.yaml" command="load"/>

  <!-- load the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"  output="screen" ns="/wpr1"  args=
  "joint_state_controller
    base_to_torso_position_controller
    torso_to_upperarm_position_controller
    upperarm_to_forearm_position_controller
    forearm_to_palm_position_controller
    palm_left_finger_position_controller
    left_finger_tip_position_controller
    palm_right_finger_position_controller
    right_finger_tip_position_controller"/>

  <!-- convert joint states to TF transforms for rviz, etc -->
  <param name="wpr1_description" textfile="$(find wpr_simulation)/models/wpr1.model" />
  <node name="gazebo_wpr1_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
    <remap from="robot_description" to="wpr1_description" />
    <remap from="/joint_states" to="/wpr1/joint_states" />
  </node>

  <node name="wpr1_sim" pkg="wpr_simulation" type="wpr1_sim" output="screen"/>
  
</launch>